File: osmToolsBasic/src/main/java/com/asamm/osmTools/cmdCommands/CmdOgr.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.asamm.osmTools.cmdCommands;

import com.asamm.osmTools.Parameters;
import com.asamm.osmTools.mapConfig.ItemMap;

import java.io.File;

/**
 *
 * @author volda
 */
public class CmdOgr extends Cmd{
    
    ItemMap map;
    
    public CmdOgr(ItemMap map) {
        super(ExternalApp.NO_EXTERNAL_APP);
        
        this.map = map;
        
        // test if shpfile for extracting exist
        //this.map  =  map;
        if (!new File(Parameters.getCoastlineShpFile()).exists()){
            throw  new IllegalArgumentException("Shapefile with world polygons "
                    + Parameters.getCoastlineShpFile() +" does not exist");
        }

        // add base parameter
        addCommand(Parameters.getOgr2ogr());
    }
    
    public void createCmd(){
        //ogr2ogr -clipsrc 14.0 35.7 14.66 36.2   malta.shp water_polygons.shp
        addCommand("-clipsrc");

        addCommand(Double.toString(map.getBoundary().getMinLon()));
        addCommand(Double.toString(map.getBoundary().getMinLat()));
        addCommand(Double.toString(map.getBoundary().getMaxLon()));
        addCommand(Double.toString(map.getBoundary().getMaxLat()));
        addCommand(map.getPathShp());
        addCommand(Parameters.getCoastlineShpFile());
        addCommand("-skipfailures");
    }
}

File: osmToolsBasic/src/main/java/com/asamm/osmTools/cmdCommands/CmdOgr.kt
package com.asamm.osmTools.cmdCommands

import com.asamm.osmTools.config.AppConfig
import com.asamm.osmTools.mapConfig.ItemMap

class CmdOgr : Cmd(ExternalApp.OGR2OGR) {

    fun clipGlobalLandPolyToMapBounds(map: ItemMap) {
        require(AppConfig.config.coastlineConfig.landPolygonShp.toFile().exists()) {
            "Shapefile with world polygons ${AppConfig.config.coastlineConfig.landPolygonShp} does not exist"
        }
        // ogr2ogr -clipsrc <minLon> <minLat> <maxLon> <maxLat> <output.shp> <input.shp> -skipfailures
        builder()
            .add("-clipsrc")
            .add(
                map.boundary.minLon.toString(),
                map.boundary.minLat.toString(),
                map.boundary.maxLon.toString(),
                map.boundary.maxLat.toString()
            )
            .add(map.pathShp.toString())
            .add(AppConfig.config.coastlineConfig.landPolygonShp.toString())
            .add("-skipfailures")
            .execute()
    }
}